ceph/datadog_checks/ceph/ceph.py

```diff
@@ -94,8 +94,8 @@ def _extract_metrics(self, raw, tags):
                 local_tags = tags + ['ceph_osd:osd%s' % osdperf['id']]
                 self._publish(osdperf, self.gauge, ['perf_stats', 'apply_latency_ms'], local_tags)
                 self._publish(osdperf, self.gauge, ['perf_stats', 'commit_latency_ms'], local_tags)
-        except KeyError:
-            self.log.debug('Error retrieving osdperf metrics')
+        except (KeyError, TypeError):
+            self.log.debug('Error retrieving osdperf metrics. Received {}', raw.get('osd_perf', {}))
```

Comment:
Looks like `KeyError` is here to catch that `osd_perf_infos` is not in `raw_osd_perf`. 

And `TypeError` is here to catch that `raw_osd_perf` is not none.

Should we instead check that `raw_osd_perf` is valid instead of try/except ?
```
raw_osd_perf = raw.get('osd_perf', {}).get('osdstats', raw.get('osd_perf', {}))

if `osd_perf_infos` in raw_osd_perf:
   ...
else:
   # log debug
```


WDYT?

Reply:
Or use default getters

Reply:
that was a massive rabbit hole :( https://github.com/DataDog/integrations-core/pull/6385

Reply:
I'm leaving it as it was cause it does not make sense to change it only here